File: samples/sum_of_odds/sum_of_odds.m
// sum_of_odds.m - calculate sum of odd integers

#import <Foundation/Foundation.h>
#import <Foundation/NSObject.h>

const CGFloat __version__ = 1.1;
const NSString *__date__ = @"2021-01-13";

const NSInteger up_to = 170;

@interface sum_of_odds: NSObject
{
  NSInteger sum, number_of_odds, last_odd;
  NSInteger i;
}
+ (BOOL) is_odd: (NSInteger)x;
- (id)init;
- (void) sum_up: (NSInteger)upto;
- (void) print_result;
@end

@implementation sum_of_odds: NSObject

+ (BOOL) is_odd: (NSInteger)x
{
  NSInteger remainder;

  remainder = x % 2;
  if (remainder == 0) return NO;
  else                return YES;
}

- (id)init
{
  [super init];
  sum = number_of_odds = last_odd = 0;
  return self;
}

- (void) sum_up: (NSInteger)x
{
  for (i = 0; i <= x; i++)
  {
    if ([sum_of_odds is_odd: i])
    {
      sum += i;
      number_of_odds++;
      last_odd = i;
    }
  }
}

- (void) print_result
{
  if (sum == pow(number_of_odds, 2))
  {
    if (number_of_odds == 0)
      ;
    else if (number_of_odds < 7)
      printf("%zd\n", sum);
    else
      printf("1 + 3 + 5 + ... + %zd + %zd = %zd\n",
          last_odd - 1, last_odd, sum);
  }
  else
    NSLog(@"Unexpected error: %zd\n", sum);
}

@end

int main(int argc, const char *argv[])
{
  id crunch;

  crunch = [[sum_of_odds alloc] init];
  [crunch sum_up: up_to];
  [crunch print_result];

  return EXIT_SUCCESS;
}

File: samples/sum_of_odds/sum_of_odds.rs
/* sum_of_odds.ru - calculate sum of odd integers */

const __VERSION__: f32 = 0.1;
const __DATE__: &str = "2021-02-24";

const UP_TO: i32 = 170;

fn is_odd(x: i32) -> bool
{
  let remainder = x % 2;
  if remainder == 0  { false }
  else               { true }
}

fn main()
{
  let mut sum: i32 = 0;
  let mut number_of_odds: i32 = 0;
  let mut last_odd: i32 = 0;

  for i in 0..(UP_TO + 1)
  {
    if is_odd(i)
    {
      sum += i;
      number_of_odds += 1;
      last_odd = i;
    }
  }

  if sum == number_of_odds.pow(2)
  {
    if number_of_odds == 0
    { }
    else if number_of_odds < 7
    { println!("{}", sum) }
    else
    {
      println!("1 + 3 + 5 + ... + {} + {} = {}",
               last_odd - 1, last_odd, sum)
    }
  }
  else
  { eprintln!("Unexpected error: {}", sum) }
}
